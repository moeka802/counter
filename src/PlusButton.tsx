// src/PlusButton.tsx
import React from 'react';
import {Text, TouchableOpacity, StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#008080',
    width: 50,
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonText: {
    color: 'white',
    fontSize: 20,
  },
});

interface Props {
  count: number;
  setCounter: (count: number) => void;
}

export default function PlusButton(props: Props) {
  const {count, setCounter} = props;
  const reduceCounter = React.useCallback(() => {
    setCounter(count + 1);
  }, [count, setCounter]);
  return (
    <TouchableOpacity style={styles.button} onPress={reduceCounter}>
      <Text style={styles.buttonText}>+</Text>
    </TouchableOpacity>
  );
}
