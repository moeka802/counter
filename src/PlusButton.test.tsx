// src/PlusButton.test.tsx
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';
import PlusButton from '../src/PlusButton';

describe('PlusButton', () => {
  const count = 10;
  it('通常のReduceButtonを描画できる', () => {
    const component = renderer.create(
      <PlusButton count={count} setCounter={() => {}} />,
    );
    expect(component).toMatchSnapshot();
  });
  it(`ボタン押下でカウントが${count}から${count + 1}になる`, () => {
    const onPressEvent = jest.fn();
    const {container} = render(
      <PlusButton count={count} setCounter={onPressEvent} />,
    );
    fireEvent.press(container.children[0]);
    expect(onPressEvent).toBeCalled();
    expect(onPressEvent.mock.calls[0][0]).toBe(count + 1);
  });
});
