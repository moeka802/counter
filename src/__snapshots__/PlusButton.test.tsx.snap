// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlusButton 通常のReduceButtonを描画できる 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#008080",
      "height": 50,
      "justifyContent": "center",
      "width": 50,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
      }
    }
  >
    +
  </Text>
</TouchableOpacity>
`;
